Source/WebCore/style/StyleTreeResolver.cpp

```diff
@@ -194,11 +194,25 @@ ResolvedStyle TreeResolver::styleForStyleable(const Styleable& styleable, Resolu
         adjuster.adjust(*style);
     }
 
-    return {
+    ResolvedStyle resolvedStyle {
         .style = WTFMove(style),
         .relations = { },
         .matchResult = WTFMove(unadjustedStyle.matchResult)
     };
+
+    if (existingStyle && resolvedStyle.style) {
+        bool positionTryUnchanged =
+            (existingStyle->positionTryFallbacks() == resolvedStyle.style->positionTryFallbacks())
+            && (existingStyle->positionTryOrder() == resolvedStyle.style->positionTryOrder());
+
+        if (positionTryUnchanged && existingStyle->usedPositionTryFallback()) {
+            auto styleWithAppliedFallback = generatePositionOption(*existingStyle->usedPositionTryFallback(), resolvedStyle, styleable, resolutionContext);
+            if (styleWithAppliedFallback)
+                resolvedStyle.style = WTFMove(styleWithAppliedFallback);
+        }
+    }
```

Comment:
I don't quite understand the logic of this patch. The idea of the existing code was that after resolving style we generate all the position options (`generatePositionOptionsIfNeeded`), then do rounds of interleaving and until we pick one (in `tryChoosePositionOption`) to get the final style. 

As I understand what we want to do here is to remember across style resolution which option we picked last time and then using that option as the first choice after style resolution as long as options are unchanged. 

However the patch adds new places where we generate a single position options and I don't quite understand why.